ui/apps/platform/src/Containers/Collections/converter.ts

```diff
@@ -25,12 +25,29 @@ const fieldToEntityMap: Record<SelectorField, SelectorEntityType> = {
 
 const LABEL_SEPARATOR = '=';
 
+export type CollectionParseError = {
```

Comment:
The `AggregateError` was a little clunky to work with so I replaced it with a simple custom type to track error messages.

(`AggregateError` is not generic and its `errors` property  was always `any[]` so this helps avoid runtime checks everywhere else the error is used.)

Reply:
Thank you for another TypeScript lesson!

The non-empty requirement in `CollectionParseError` type and predicate function makes me curious to ask whether TypeScript reports an error if `parseCollection` has an incorrect code path which returns an empty `errors` array?